tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/MetaAttributeGenerationVisitor.java

```diff
@@ -138,6 +138,16 @@ private AnnotationMetaAttribute createMetaCollectionAttribute(DeclaredType decla
 				accessTypeInfo.setDefaultAccessType( entity.getEntityAccessTypeInfo().getAccessType() );
 			}
 		}
+		if (( TypeUtils.containsAnnotation( element, Constants.CONVERT ) ||
+				TypeUtils.containsAnnotation( element, Constants.HIBERNATE_TYPE ) ) &&
+				!isCollectionAssociation( element )
+		) {
```

Comment:
One-shot usage of `TypeUtils.containsAnnotation(Element, String)` is okay, but multiple invocations won't be as efficient as
```
AnnotationMirror annotationMirrors =  element.getAnnotationMirrors();
// go about checking based on `AnnotationMirror` using `public static boolean isAnnotationMirrorOfType(AnnotationMirror annotationMirror, String fqcn)` method
```

Furthermore, I do think we can reformat the above code in some way to improve readability. For instance, I spent quite some time figuring out how the parenthesis pairs match up above.

Reply:
Sure thing. Feel free to modify it any way you want and push the commit.

Reply:
I don't think I have such rights, but I can create a PR after yours is merged.

Reply:
I think you can [fork my repository](https://github.com/vladmihalcea/hibernate-orm/tree/HHH-12338), and your change and send a Pull Request.

Reply:
Thanks for the tip, but I would approve the PR if you made the code cleaning as I proposed, :).

Reply:
I don't know exactly what changes you want to make, so it's better if you cherry-pick my commit, add your changes and submit a new PR to the main fork.

I'll close my PR once you do that.

Reply:
I think 
```
if (( TypeUtils.containsAnnotation( element, Constants.CONVERT ) ||
				TypeUtils.containsAnnotation( element, Constants.HIBERNATE_TYPE ) ) &&
				!isCollectionAssociation( element )
		) {
```
could be simplified dramatically by
```
if ( TypeUtils.containsAnnotation( element, Constants.CONVERT, Constants.HIBERNATE_TYPE ) && ! TypeUtils.containsAnnotation( element, Constants.ONE_TO_MANY, Constants.MANY_TO_MANY, Constants.ELEMENT_COLLECTION) ) {

```

Reply:
@NathanQingyangXu I pushed a new commit based on your changes.

Reply:
thanks. I approved. I would let other reviewer focus on business logic stuff.